test/Riok.Mapperly.Tests/Mapping/EnumerableTest.cs

```diff
@@ -631,6 +631,47 @@ public void MapToExistingExplicitAddCustomCollection()
             .HaveSingleMethodBody("");
     }
 
+    [Fact]
+    public void MapToExistingExplicitCountCustomCollection()
+    {
+        // should not create an EnsureCapacity statement when the target has an explicit count
+        var source = TestSourceBuilder.MapperWithBodyAndTypes(
+            "partial void Map(IEnumerable<int> source, B target);",
+            "class B : List<int> { int List<int>.Count { get; } }");
```

Comment:
I don't get this test case, class B extends the List class, but implements the base class property explicit. This does not really work, does it?

Reply:
Yeah this test is terrible for many reasons. I decided not to generate an EnsureCapacity statement when the target type has an `EnsureCapacity` method and an explicit length. This is because I deemed it to be an edge case and a complete pain to handle with `EnsureCapacityIsType`. 

For some reasons I made a (broken) test to ensure it won't generate an `EnsureCapacity` statement under these conditions.

I could either delete this test, fix it, or update the logic to insert a cast to cast the target when its `Count` is explicit.

Reply:
I'm okay with deleting this test.